packages/aragon-wrapper/src/index.js

```diff
@@ -450,6 +469,276 @@ export default class Aragon {
       publishReplay(1)
     )
     this.apps.connect()
+
+    /*******************************
+     *                             *
+     *            REPOS            *
+     *                             *
+     ******************************/
+
+    // Initialize installed repos from the list of apps
+    const installedRepoCache = new Map()
+    const repo$ = apps$.pipe(
+      // Map installed apps into a deduped list of their aragonPM repos, with these assumptions:
+      //   - No apps are lying about their appId (malicious apps _could_ masquerade as other
+      //     apps by setting this value themselves)
+      //   - `contractAddress`s will stay the same across all installed apps.
```

Comment:
What is `contractAddress`? how could it stay the same for different apps (respectively different APM repo)?

Reply:
Yeah, `codeAddress` and `contractAddress` are almost synonymous terms that could be explained a bit more.

- `codeAddress` is used by the **proxy** contract, to describe where its implementation lives
- `contractAddress` is used by the **Repo** contract, to declare a specific version's on-chain package location

A **proxy**'s `codeAddress` should always be pointing (linking) to a **Repo** version's `contractAddress`. An upgrade to the **proxy** will modify its `codeAddress` to point to another version's `contractAddress`.